### PR TITLE
[Image Beta] port download image feature

### PR DESCRIPTION
**Pull request by jtbandes**
**User-Facing Changes**
None

**Description**
Adds Download button to object inspector.
image

### CODE REVIEW

**Review by snosenzo — APPROVED** [review]
looks good if we want to do it this way but feels a bit strange the way we're doing it

**Review by snosenzo — APPROVED** [review]
I think we just need to remove the canvas from the dom after downloading

File: packages/studio-base/src/panels/ThreeDeeRender/renderables/Images/ImageRenderable.ts [hunk 1]
```diff
@@ -302,12 +293,62 @@ export class ImageRenderable extends Renderable<ImageUserData> {
...
+      } else {
+        const imageData = new ImageData(image.width, image.height);
+        decodeRawImage(image, {}, imageData.data);
+        bitmap = await createImageBitmap(imageData);
+      }
+
+      const width = rotation === 90 || rotation === 270 ? bitmap.height : bitmap.width;
+      const height = rotation === 90 || rotation === 270 ? bitmap.width : bitmap.height;
+
+      const canvas = document.createElement("canvas");
```
[the remark this answers is not in the document]
Comment (jtbandes) [reply packages/studio-base/src/panels/ThreeDeeRender/renderables/Images/ImageRenderable.ts:319]: It's never added to the body

**Comment by jtbandes** [comment]
Product review feedback: this should be in the right click menu, or deleted

**Pull request merged**